src/he/conn.c

```diff
@@ -416,12 +416,19 @@ he_return_code_t he_internal_send_message(he_conn_t *conn, uint8_t *message, uin
   // For now - this will just do a simple write and assume it works
   int res = wolfSSL_write(conn->wolf_ssl, message, (int)length);
 
-  // Note that we expect wolfSSL_write callback to ALWAYS succeed
-  // Any output buffering is done by the host app
-  if(res == 0) {
-    return HE_ERR_CONNECTION_WAS_CLOSED;
-  } else if(res < 0) {
-    return HE_ERR_SSL_ERROR;
+  if(res <= 0) {
+    int error = wolfSSL_get_error(conn->wolf_ssl, res);
```

Comment:
Can we get some tests to test this new behaviour?

Reply:
The test is already updated to ensure `wolfSSL_get_error` is called.

Reply:
The test https://github.com/expressvpn/lightway-core/blob/5166e138e5562f613771459d70a3f66af98bed77/test/he/test_conn.c#L592-L597 is quite lacking. I think its a good opportunity to improve the test suite

Reply:
Done!